sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -593,6 +593,11 @@ object GpuOverrides extends Logging {
     lit.value == null
   }
 
+  def isSupportedStringReplacePattern(strLit: String): Boolean = {
+    // check for regex special characters, except for \u0000 which we can support
+    !regexList.filterNot(_ == "\u0000").exists(pattern => strLit.contains(pattern))
```

Comment:
I think it's more readable in a conjunctive form, but it is not part of your PR, so very optional:
```suggestion
    !regexList.exists(pattern => pattern != "\u0000" && strLit.contains(pattern))
```